packages/react/src/primitives/types/button.ts

```diff
@@ -11,6 +11,13 @@ export type ButtonVariations =
   | 'warning'
   | 'destructive';
 
+export type ButtonColorTheme =
+  | 'info'
+  | 'error'
+  | 'warning'
```

Comment:
This might just be a me problem, but having a hard time understanding why there is both a `warning` variant and a `warning` color scheme? Is this just to prevent a breaking change for `variation`?

Reply:
This is one of the things that spurred this refresh; there was a `warning` and `destructive` variation added for Button for Account Settings. They are both red (which sort of deviates from our pattern of using orange for warning like in Alert/Message). Additionally, having variations that are really just color modifications from an existing variation is a bit limiting for adding future color modifications (like what do we call a variation that is a green button that has a solid background color vs a green button that has an outline?). With this change, we have to worry less about naming and have more options for buttons where a color scheme is useful (i.e. buttons that might have an action associated with them like "approve" or "delete") or where they can aid with better contrast (i.e. our Alert/Message primitive has a colored background which can clash with a branded color theme like our teal buttons have). 

This isn't in this PR, but ideally, we should deprecate the `warning` and `destructive` variations and customers would use:
```
<Button variation="primary" colorTheme="error" />
```
to achieve the current "destructive" style. And they would use:
```
<Button colorTheme="error" />
```
to achieve the existing "warning" style (which is really another red danger button).

Let me know if that helps clarify!

Reply:
IMO `error` means something has already gone wrong (which might mean it should be scoped to "message" related components), `warning` should warn the user, and `destructive` should indicate an action that is well, _destructive_ hehe. Totally onboard with removing them as variations but do think they have semantic meaning that `error` does not convey. Kind of wonder if we should include `destructive` as a `colorTheme` and consider removing `error`?

Reply:
I actually considered using color namespaces from our theme (red, green, blue, purple, etc) as names for the colorThemes and foregoing any association with semantics/action but during the doc review we settled on this (using just warning, success, info, error) as a more limited start. Perhaps it could be expanded in the future to support any namespace from our color palette, and these end up just being special "named" color themes that inherit those base not-special color themes

Generally, I don't want to actually tie too much semantic meaning to a name of a color scheme; but these are so common across design systems that I think it's ok. Especially if we're already associating these names with colors elsewhere in our own design system and reinforcing that pattern. Happy to be convinced though if we want to chat offline with Danny!